app/lib/project/MetapolatorProject.js

```diff
@@ -614,6 +616,38 @@ define([
         exportController.export();
     };
 
+    _p.exportZipInstance = function(masterName, instanceName, precision) {
+        //export the font to a temporary directory
+        var temp_dir = instanceName+"_temp";
+        this.exportInstance(masterName, temp_dir, precision);
```

Comment:
instead of writing to a `temp_dir` on the current io module we should find a way to write to a new "InMemory" io module. It's possible that the io module is a REST interface that sends io via the network, this would make your code perform incredibly bad.

For this we probably have to change the `exportInstance` method but the good news is that we can remove your inline version of `rmDirRecursive`